grafana-plugin/src/pages/integration_2/Integration2.tsx

```diff
@@ -579,10 +579,14 @@ class Integration2 extends React.Component<Integration2Props, Integration2State>
   getTemplatesList = (): CascaderOption[] => INTEGRATION_TEMPLATES_LIST;
 
   openEditTemplateModal = (templateName, channelFilterId?: ChannelFilter['id']) => {
-    this.setState({
-      isEditTemplateModalOpen: true,
-      selectedTemplate: templateForEdit[templateName],
-    });
+    if (templateForEdit[templateName]) {
+      this.setState({
+        isEditTemplateModalOpen: true,
+        selectedTemplate: templateForEdit[templateName],
+      });
+    } else {
+      openErrorNotification('Template can not be edited. Please contact support.');
```

Comment:
How can we reach this state where the template is undefined?

Reply:
For example if the name of the template will be changed from backend, and we did not change it on frontend side, because I am using TemplateForEdit config to have all the parameters I need
Also we have this problem when we update in OSS version and forget smth in private (happened twice already)